openshift/system/Dockerfile

```diff
@@ -1,19 +1,20 @@
-FROM quay.io/centos/centos:stream8 as builder
+FROM quay.io/centos/centos:stream9 as builder
 WORKDIR /opt/system
-ENV RUBY_MAJOR_VERSION=2 \
-    RUBY_MINOR_VERSION=7 \
+ENV RUBY_MAJOR_VERSION=3 \
+    RUBY_MINOR_VERSION=1 \
     RAILS_ENV=production \
-    NODE_ENV=production
+    NODE_ENV=production \
+    # This is to fix 'error:0308010C:digital envelope routines::unsupported' in Node 18
+    # the proper fix should be upgrading webpack and babel-loader
+    NODE_OPTIONS='--openssl-legacy-provider'
```

Comment:
If we are always going to need `NODE_OPTIONS='--openssl-legacy-provider'` from now on to be able to compile webpack assets, why not add it to the rake task itself?

I would prefer to avoid adding this env var to my RubyMine launchers and have to remember to set it before calling the task from console.

Reply:
Yeah, makes sense (even though I think we'll be able to get rid of it soon), although I'm not sure which task it should be added to :grimacing:

Reply:
> the proper fix should be upgrading webpack and babel-loader

This sounds promising. Maybe put that option behind a check of webpack version (if that is feasible). Just a suggestion if it makes sense to you.

Reply:
> I'm not sure which task it should be added to

Mmm, can we override a rake task to set the variable and call `super`? Is that legal? 🤔

Reply:
I don't think so, but this seems to be working: https://github.com/3scale/porta/pull/3847/commits/2edb2c3340bd63523310e2a75192a1ce1b18fa6f
Now, probably this affects other tasks also... probably `webpack:dev`

Reply:
It doesn't work for me, I still see the error 😔

> Now, probably this affects other tasks also... probably webpack:dev

I think you only need to `enhance` the task `webpacker:compile` rather than `assets:precompile`, that way it wouldn't affect any other task AFAIK.

Reply:
> It doesn't work for me, I still see the error 😔
> 
> > Now, probably this affects other tasks also... probably webpack:dev
> 
> I think you only need to `enhance` the task `webpacker:compile` rather than `assets:precompile`, that way it wouldn't affect any other task AFAIK.

I am not sure about which task needs to be enhanced to be honest... but it seems that `webpack:dev` doesn't depend on any of those tasks anyway, so I also added https://github.com/3scale/porta/pull/3847/commits/9be727755d4bfd98e77b3c0530e35e8609283484

Reply:
That works perfect 👍